flakiness in tests/views/test_response_correlation.py::test_show_respo_with_obs

(partial) jenkins output:

```
=================================== FAILURES ===================================
___________________________ test_show_respo_with_obs ___________________________

    def test_show_respo_with_obs(mock_data, dash_duo):
        """Test response observation filter works as expected"""

        ...
    
        expected_def_responses = ["FGPT", "FOPR", "SNAKE_OIL_GPR_DIFF", "WOPR:OP1"]
        assert default_responses == expected_def_responses
    
        expected_obs_responses = ["FOPR", "WOPR:OP1"]
        obs_radio_btn = dash_duo.find_element(
            "#" + plugin.uuid("response-observations-check")
        )
        obs_radio_btn.click()
        obs_responses = get_options(dash_duo, "#" + resp_selector_id)
>       assert obs_responses == expected_obs_responses
E       AssertionError: assert ['FGPT', 'FOP...', 'WOPR:OP1'] == ['FOPR', 'WOPR:OP1']
E         At index 0 diff: 'FGPT' != 'FOPR'
E         Left contains 2 more items, first extra item: 'SNAKE_OIL_GPR_DIFF'
E         Full diff:
E         - ['FOPR', 'WOPR:OP1']
E         + ['FGPT', 'FOPR', 'SNAKE_OIL_GPR_DIFF', 'WOPR:OP1']

tests/views/test_response_correlation.py:133: AssertionError
```

looks to me like the button press was not processed.

--- a/tests/views/test_response_correlation.py
+++ b/tests/views/test_response_correlation.py
@@ -76,21 +76,20 @@ def test_axes_labels(mock_data, dash_duo):
         response_selector_id = plugin.uuid("parameter-selector-multi-resp")
         select_by_name(dash_duo, f"#{response_selector_id}", wanted_response)
 
-        # wait a bit for the graph to be drawn
-        wait_a_bit(dash_duo)
-
         plot_id = plugin.uuid("response-overview")
 
         # check that y axis label spells out "Value"
-        y_axis_title = dash_duo.find_element(f"#{plot_id} text.ytitle")
-        assert y_axis_title.text == "Value"
+        dash_duo.wait_for_text_to_equal(f"#{plot_id} text.ytitle", "Value", timeout=5)
 
         # check that one has date, the other has index as x axis label
-        x_axis_title = dash_duo.find_element(f"#{plot_id} text.xtitle")
         if wanted_response == "FOPR":
-            assert x_axis_title.text == "Date"
+            dash_duo.wait_for_text_to_equal(
+                f"#{plot_id} text.xtitle", "Date", timeout=5
+            )
         else:
-            assert x_axis_title.text == "Index"
+            dash_duo.wait_for_text_to_equal(
+                f"#{plot_id} text.xtitle", "Index", timeout=5
+            )
 
         # clear response selection so next selection can be displayed
         response_deactivator_id = plugin.uuid("parameter-deactivator-resp")

--- a/tests/views/test_response_correlation.py
+++ b/tests/views/test_response_correlation.py
@@ -5,7 +5,6 @@ from tests.conftest import (
     select_by_name,
     select_ensemble,
     wait_a_bit,
-    get_options,
 )
 
 
@@ -117,16 +116,20 @@ def test_show_respo_with_obs(mock_data, dash_duo):
 
     select_ensemble(dash_duo, plugin, "default3")
 
-    resp_selector_id = plugin.uuid("parameter-selector-multi-resp")
-    default_responses = get_options(dash_duo, "#" + resp_selector_id)
+    expected_default_responses = ["FGPT", "FOPR", "SNAKE_OIL_GPR_DIFF", "WOPR:OP1"]
+    expected_responses_with_observations = ["FOPR", "WOPR:OP1"]
 
-    expected_def_responses = ["FGPT", "FOPR", "SNAKE_OIL_GPR_DIFF", "WOPR:OP1"]
-    assert default_responses == expected_def_responses
+    response_selector_id = "#" + plugin.uuid("parameter-selector-multi-resp")
 
-    expected_obs_responses = ["FOPR", "WOPR:OP1"]
+    dash_duo.wait_for_text_to_equal(
+        response_selector_id, "\n".join(expected_default_responses)
+    )
+
     obs_radio_btn = dash_duo.find_element(
         "#" + plugin.uuid("response-observations-check")
     )
     obs_radio_btn.click()
-    obs_responses = get_options(dash_duo, "#" + resp_selector_id)
-    assert obs_responses == expected_obs_responses
+
+    dash_duo.wait_for_text_to_equal(
+        response_selector_id, "\n".join(expected_responses_with_observations)
+    )

